Tests/Resources/ExpectedSwiftCode/Responses/MultipleFragmentsQueryResponseNext.swift

```diff
@@ -61,34 +61,16 @@ struct MultipleFragmentsQueryResponse: GraphApiResponse, Equatable {
 						self.__typename = "CustomerEdge"
 				}
 					// MARK: - Nested Types
-						public struct Node: GraphApiResponse, Equatable {
+						@dynamicMemberLookup
+				public struct Node: GraphApiResponse, Equatable {
 					// MARK: - Response Fields
-						/// Globally unique identifier.
-						public var id: GraphID {
-							get {
-								return asBasicFragmentFragment.id
-							}
-							set {
-								asBasicFragmentFragment.id = newValue
-							}
+						public subscript<T>(dynamicMember keyPath: WritableKeyPath<MerchantApi.BasicFragment, T>) -> T {
+							get { asBasicFragmentFragment[keyPath: keyPath] }
+							set { asBasicFragmentFragment[keyPath: keyPath] = newValue }
 						}
-						/// A note about the customer.
-						public var note: String? {
-							get {
-								return asBasicFragmentFragment.note
-							}
-							set {
-								asBasicFragmentFragment.note = newValue
-							}
-						}
-						/// The customer's last order.
-						public var lastOrder: MerchantApi.MultiLevelFragment.LastOrder? {
-							get {
-								return asMultiLevelFragmentFragment.lastOrder
-							}
-							set {
-								asMultiLevelFragmentFragment.lastOrder = newValue
-							}
+						public subscript<T>(dynamicMember keyPath: WritableKeyPath<MerchantApi.MultiLevelFragment, T>) -> T {
+							get { asMultiLevelFragmentFragment[keyPath: keyPath] }
+							set { asMultiLevelFragmentFragment[keyPath: keyPath] = newValue }
```

Comment:
Does this address your comment here?:  https://github.com/Shopify/syrup/pull/1/files#r391246173

Reply:
Yeah, that looks like its would be alright. I'm still a bit worried that we're ignoring computedFragmentFields now. I would suggest looking into the source to figure out how it gets its values. If it is just a flattened array of all fields used by a fragment, then we should be fine.

Reply:
I don't understand what you mean.  Are you saying that the change may ignore computed fragment's computed fragments?  For example, what would happen if

```graphql
query MultipleFragmentsQuery {
  shop {
    customers(first: 10) {
      edges {
        node {
          ...BasicFragment
          ...MultiLevelFragment
          ... on Visit {
            location
            time
            place
        }
      }
    }
  }
}
```
or something similar?

Reply:
There is logic inside rocky to determine the computedFragmentFields [here](https://github.com/Shopify/syrup/blob/master/Sources/SyrupCore/IR/IntermediateRepresentation%2BCollectedField.swift#L737-L759). In that code, there is some logic to make sure the fields within the fragment actually exist on the parent type. I believe this is the case for handling something like this:
```graphql
fragment ProductFields on Node {
  ... on Product {
    title
    price
  }
}

query NodeQuery {
  node(id: ...) {
    ... ProductFields
  }
}
```

In this example, I'm pretty sure computedFragmentFields should filter out the product specific fields, since the parent type of the fields in the ProductFields fragment does not match the Node interface type.

If we are suddenly ignoring the computedFragmentFields property, I think we expose ourselves to the same kind of bugs that it tries to prevent.